pkgs/build-support/appimage/default.nix

```diff
@@ -55,12 +63,33 @@ rec {
       meta = {
         sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
       } // meta;
-    } // (removeAttrs args ([ "pname" "version" ] ++ (builtins.attrNames (builtins.functionArgs wrapAppImage)))));
-
-  wrapType2 = args@{ name ? "${args.pname}-${args.version}", src, extraPkgs ? pkgs: [ ], ... }: wrapAppImage
-    (args // {
+    } // (removeAttrs args ([ "pname" "version" ] ++ (lib.attrNames (builtins.functionArgs wrapAppImage)))));
+
+  wrapType2 =
+    { name ? "${args.pname}-${args.version}"
+    , src
+    , extraPkgs ? pkgs: [ ]
+    , extraInstall ? extracted: ""
+    , ...
+    } @ args:
+    let
+      extracted = extract { inherit name src; };
+      extraInstallString = extraInstall extracted;
+    in
+    wrapAppImage (args // {
       inherit name extraPkgs;
-      src = extract { inherit name src; };
+      src = extracted;
+
+      extraInstallCommands = args.extraInstallCommands or ''
+        # Dont include the version number in the binary name
+        if [ -f "$out/bin/${name}" ]; then
+          mv "$out/bin/${name}" "$out/bin/${lib.getName name}"
+        fi
+
+        # Appimages always have a desktop file, copy it to our output
+        mkdir -p $out/share/applications
+        cp ${extracted}/*.desktop $out/share/applications
+      '' + lib.optionalString (extraInstallString != "") extraInstallString;
```

Comment:
I think extraInstallCommands should be append by default so that extraInstallString can be removed and to overwrite it you specific installCommands.

Reply:
Note quite sure what you mean with "to overwrite it you specific installCommands". `extraInstall` defaults to an emtpy string already, do we need a way to override that?

I've appended it by default now and removed `extraInstallString`.